src/main/java/com/marklogic/kafka/connect/sink/DefaultSinkRecordConverter.java

```diff
@@ -73,7 +105,29 @@ protected AbstractWriteHandle toContent(SinkRecord record) {
 			throw new NullPointerException("'record' must not be null, and must have a value.");
 		}
 		Object value = record.value();
+		Schema schema = record.valueSchema();
+		/* Determine the converter */
+		if (schema != null && value instanceof Struct) {
+			/* Avro, ProtoBuf or JSON with schema, ignore schema, handle only the value */
+			converter = "AVRO_OR_JSON_WITH_SCHEMA";
+			logger.info("Avro or JsonWithSchema document received. Converting to byte[].");
+			final String payload = new String(JSON_CONVERTER.fromConnectData(record.topic(), schema, value), StandardCharsets.UTF_8);
+			value = payload.getBytes();
+		}
+		
+        if (value instanceof Map) {
```

Comment:
This is another instance where a Strategy interface could be used, as it's getting a bit more complicated - gotta check for not just a byte[], but a Struct and a Map. 

How much of a connection is there between these strategies and the ones for ContentIdExtractor? I am wondering if we need a separate interface here - e.g. ContentExtractor - which does "AbstractWriteHandle toContent(SinkRecord)" - and the implementations of that may implement ContentIdExtractor as well.